### PR TITLE
fix(truncate): reject relative size overflow

### PR DESCRIPTION
**Pull request by ThomasHabets**
Relative extensions used unchecked u64 addition, so optimized builds could wrap the target size and truncate data. Debug builds panic.

This fix is in line with how the C `truncate` already works.

This also fixes the error message for `round up`. Though that's even more of an edge case.

Before:

```
$ truncate -s 1 file
$ truncate -s +18446744073709551615 file
$ test -s file || file is empty
file is empty
$
```

file succeeded and shrank file to 0 bytes.

After: the second command fails and leaves file at 1 byte.

```
$ truncate -s 1 file
$ truncate -s +18446744073709551615 file
truncate: Invalid number: '+18446744073709551615': Value too large for defined data type
$ test -s file || file is empty
$
```

Found with LLM.

### CODE REVIEW

**Comment by ThomasHabets** [comment]
> GNU testsuite comparison:
[…]

I don't believe any of that is related to this chang.

File: src/uu/truncate/src/truncate.rs [hunk 1]
```diff
@@ -230,7 +249,26 @@ fn file_truncate(
...
+        .map_err(|error| match error {
+            SizeCalculationError::DivisionByZero => {
+                USimpleError::new(1, translate!("truncate-error-division-by-zero"))
+            }
+            SizeCalculationError::Overflow => {
+                let error = size_argument.map_or_else(
+                    || "Value too large for defined data type".to_string(),
+                    |argument| {
+                        format!(
+                            "{}: Value too large for defined data type",
```
Comment (cakebaker) [inline src/uu/truncate/src/truncate.rs]: Can you please use the `translate!` macro for the strings?

**Review by cakebaker — COMMENTED** [review]
The other changes look fine :)

File: src/uu/truncate/src/truncate.rs (hunk 1 above)
Comment (ThomasHabets) [reply src/uu/truncate/src/truncate.rs]: Done.

I don't speak french, but copy-pasted the message from `src/uu/stdbuf/locales/en-US.ftl`.
